virtio-devices/src/block.rs

```diff
@@ -660,6 +661,7 @@ impl Block {
                     physical_block_exp,
                     min_io_size: (topology.minimum_io_size / logical_block_size) as u16,
                     opt_io_size: (topology.optimal_io_size / logical_block_size) as u32,
+                    seg_max: std::cmp::max(queue_size as u32, 3) - 2,
```

Comment:
How about check queue_size must >= 2 in Diskconfig::validate()？

For example, in qemu：
https://gitlab.com/qemu-project/qemu/-/blob/master/hw/block/virtio-blk.c?ref_type=heads#L1846

Reply:
Sure, this is fine. I'm working on fixing the vhdx error at the moment.